src/components/asset-list/RecyclerAssetList2/profile-header/ProfileActionButtonsRow.tsx

```diff
@@ -0,0 +1,426 @@
+  useSwapCurrencyHandlers,
+  useWallets,
+} from '@/hooks';
+import { delayNext } from '@/hooks/useMagicAutofocus';
+import { useNavigation } from '@/navigation';
+import { useTheme } from '@/theme';
+import { showActionSheetWithOptions, watchingAlert } from '@/utils';
+import Routes from '@rainbow-me/routes';
+import { FloatingEmojis } from '@/components/floating-emojis';
+import showWalletErrorAlert from '@/helpers/support';
+import { analytics } from '@/analytics';
+import ContextMenu from '@/components/native-context-menu/contextMenu';
+import { useRecoilState } from 'recoil';
+import { addressCopiedToastAtom } from '@/screens/WalletScreen';
+import config from '@/model/config';
+
+export const ProfileActionButtonsRowHeight = 80;
+
+export function ProfileActionButtonsRow() {
+  // ////////////////////////////////////////////////////
+  // Account
+  const { accountColor, accountImage } = useAccountProfile();
+
+  // ////////////////////////////////////////////////////
+  // Colors
+
+  const { result: dominantColor, state } = usePersistentDominantColorFromImage(
+    maybeSignUri(accountImage ?? '') ?? ''
+  );
+
+  const { colors } = useTheme();
+  let accentColor = colors.white;
+  if (accountImage) {
+    accentColor = dominantColor || colors.white;
+  } else if (typeof accountColor === 'number') {
+    accentColor = colors.avatarBackgrounds[accountColor];
+  }
+
+  // ////////////////////////////////////////////////////
+  // Animations
+
+  const hasAvatarLoaded = !accountImage && accountColor !== undefined;
+  const hasImageColorLoaded = state === 2 || state === 3;
+  const hasLoaded = hasAvatarLoaded || hasImageColorLoaded;
+
+  const scale = useDerivedValue(() => (hasLoaded ? 1 : 0.9));
+  const expandStyle = useAnimatedStyle(() => ({
+    transform: [
+      {
+        scale: withSpring(scale.value, {
+          damping: 12,
+          restDisplacementThreshold: 0.001,
+          restSpeedThreshold: 0.001,
+          stiffness: 280,
+        }),
+      },
+    ],
+  }));
+
+  if (!hasLoaded) return null;
+  return (
+    <Box width="full">
+      <Inset horizontal={{ custom: 17 }}>
+        <AccentColorProvider color={accentColor}>
+          <Columns>
+            <Column>
+              <Animated.View style={[expandStyle]}>
+                <CopyButton />
+              </Animated.View>
+            </Column>
+            <Column>
+              <Animated.View style={[expandStyle]}>
+                <SwapButton />
+              </Animated.View>
+            </Column>
+            <Column>
+              <Animated.View style={[expandStyle]}>
+                <SendButton />
+              </Animated.View>
+            </Column>
+            <Column>
+              <Animated.View style={[expandStyle]}>
+                <MoreButton />
+              </Animated.View>
+            </Column>
+          </Columns>
+        </AccentColorProvider>
+      </Inset>
+    </Box>
+  );
+}
+
+function ActionButton({
```

Comment:
Could this component be in its own file?

Reply:
Could be! I wanted to avoid early abstraction.